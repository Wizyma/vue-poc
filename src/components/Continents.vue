<template>
  <div>
    <div v-if="continents">
      <div v-for="continent in continents" v-bind:key="continent.code">
        <div>
          <h1>{{ continent.name }}</h1>
          <div>
            <ul class="list">
              <li
                class="list-item"
                v-for="countrie in continent.countries"
                v-bind:key="countrie.code"
              >
                <p class="list-item-text">{{ countrie.name }}</p>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div v-else>Loading....</div>
  </div>
</template>

<style scoped>
.list {
  list-style: none;
  padding: 0px;
  display: flex;
  flex-grow: 1;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.list-item {
  display: flex;
  text-align: left;
  width: 200px;
  height: 200px;
  align-items: center;
  justify-content: center;
  margin: 10px;
  box-shadow: 1px 4px 14px 2px rgba(0, 0, 255, 0.2);
}

.list-item-text {
  width: 150px;
  white-space: nowrap;
  text-align: center;
  overflow: hidden;
  text-overflow: ellipsis;
}
</style>

<script lang="ts">
import Vue from "vue";
import gql from "graphql-tag";

const CONTINENTS_QUERY = gql`
  query {
    continents {
      name
      code
      countries {
        code
        name
      }
    }
  }
`;

export default {
  name: "Continents-Component",
  apollo: {
    continents: {
      query: CONTINENTS_QUERY
    }
  }
};
</script>
