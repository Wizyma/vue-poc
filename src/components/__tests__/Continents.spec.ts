import { itRenderVueGraphQLComponentCorrectly } from "../../../tests/test-utils";
import Continents from '../Continents.vue';

const mutations = [
  {
    name: "with response",
    props: {
      continents: [{
        name: "Africa",
        code: "AF",
        countries: [{
          name: "Algerie",
          code: "AL",
        }]
      }, {
        name: "Europe",
        code: "EU",
        countries: [{
          name: "France",
          code: "FR"
        }]
      }]
    }
  },
  {
    name: "without response",
    props: null
  }
];

describe(Continents.name, () => {
  itRenderVueGraphQLComponentCorrectly(Continents, mutations);
});
