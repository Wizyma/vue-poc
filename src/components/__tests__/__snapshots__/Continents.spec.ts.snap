// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Continents-Component Continents-Component with responseshould render correctly  1`] = `
<div>
  <div>
    Loading....
  </div>
</div>
`;

exports[`Continents-Component Continents-Component without responseshould render correctly  1`] = `
<div>
  <div>
    Loading....
  </div>
</div>
`;
